File: data/src/main/java/me/dewari/ajay/data/database/NoteDatabase.java
package me.dewari.ajay.data.database;

import android.content.Context;
import android.os.AsyncTask;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.sqlite.db.SupportSQLiteDatabase;

import com.binarybuff.architecturecomponentdemo.Daos.NoteDao;
import com.binarybuff.architecturecomponentdemo.Models.Note;

@Database(entities = {Note.class}, version = 1)
public abstract class NoteDatabase extends RoomDatabase {

    private static NoteDatabase instance;

    public abstract NoteDao noteDao();

    public static synchronized NoteDatabase getInstance(Context context) {
        if (null == instance) {
            instance = Room.databaseBuilder(context.getApplicationContext(), NoteDatabase.class, "note_database")
                    .fallbackToDestructiveMigration().addCallback(roomCallback).build();
        }
        return instance;
    }

    private static RoomDatabase.Callback roomCallback= new RoomDatabase.Callback(){
        @Override
        public void onCreate(@NonNull SupportSQLiteDatabase db) {
            super.onCreate(db);
            new AddDefaultNoteAsyncTask(instance).execute();
        }
    };

    private static class AddDefaultNoteAsyncTask extends AsyncTask<Void, Void, Void> {
        private NoteDao noteDao;
        public AddDefaultNoteAsyncTask(NoteDatabase database) {
            this.noteDao = database.noteDao();
        }

        @Override
        protected Void doInBackground(Void... voids) {
            noteDao.insert(new Note("Note 1", "This is 1st note ", 2));
            noteDao.insert(new Note("Note 2", "This is 1st note ", 3));
            noteDao.insert(new Note("Note 3", "This is 3rd note having the higest priority", 1));
            return null;
        }
    }
}

File: data/src/main/java/me/dewari/ajay/data/database/NoteDatabase.kt
package me.dewari.ajay.data.database

import androidx.room.Database
import androidx.room.RoomDatabase
import me.dewari.ajay.data.dao.NoteDao
import me.dewari.ajay.data.entity.Note

@Database(entities = arrayOf(Note::class), version = 1, exportSchema = false)
abstract class  NoteDatabase: RoomDatabase() {

    abstract fun wordDao(): NoteDao
    companion object {
        // Singleton prevents multiple instances of database opening at the
        // same time.
        @Volatile
        private var INSTANCE: NoteDatabase? = null

        fun getDatabase(context: Context): NoteDatabase {
            val tempInstance = INSTANCE
            if (tempInstance != null) {
                return tempInstance
            }
            synchronized(this) {
                val instance = Room.databaseBuilder(
                    context.applicationContext,
                    NoteDatabase::class.java,
                    "word_database"
                ).build()
                INSTANCE = instance
                return instance
            }
        }
    }
}